### ISSUE TITLE
[package] zlib/1.2.11: can't compile with clang on windows

### ISSUE THREAD

**Issue by AndreyMlashkin** [conan-io/conan-center-index#9806]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **zlib/1.2.11**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 16 - clang**
  * Conan version: **conan 1.46.0**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.version=12
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
```

### Steps to reproduce (Include if Applicable)
 conan install zlib/1.2.11@ --build missing -pr clang

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.version=12
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
zlib/1.2.11: WARN: Package binary is corrupted, removing: bd500648356902bb5cb5a6961daa81fc25f3f22c
Installing package: zlib/1.2.11
Requirements
    zlib/1.2.11 from 'artifactory' - Cache
Packages
    zlib/1.2.11:bd500648356902bb5cb5a6961daa81fc25f3f22c - Build

Installing (downloading, building) binaries...
zlib/1.2.11: WARN: Build folder is dirty, removing it: E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c
zlib/1.2.11: Copying sources to build folder
zlib/1.2.11: Building your package in E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c
zlib/1.2.11: Generator cmake created conanbuildinfo.cmake
zlib/1.2.11: Aggregating env generators
zlib/1.2.11: Calling build()
zlib/1.2.11: Found Visual Studio 16
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.11
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

----Running------
> cd E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c\build_subfolder && cmake -G "Ninja" -DCMAKE_BUILD_TYPE="Debug" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="clang" -DCONAN_COMPILER_VERSION="12" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="E:\conan_new\zlib\1.2.11\_\_\package\bd500648356902bb5cb5a6961daa81fc25f3f22c" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DSKIP_INSTALL_ALL="False" -DSKIP_INSTALL_LIBRARIES="False" -DSKIP_INSTALL_HEADERS="False" -DSKIP_INSTALL_FILES="True" -Wno-dev E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c
-----------------
-- The C compiler identification is Clang 12.0.0 with MSVC-like command-line
-- The CXX compiler identification is Clang 12.0.0 with MSVC-like command-line
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: E:/msvc2019/VC/Tools/Llvm/x64/bin/clang-cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: E:/msvc2019/VC/Tools/Llvm/x64/bin/clang-cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting language standard
-- Conan: Compiler Clang>=8, checking major version 12
-- Conan: Checking correct version: 12
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Renaming
--     E:/conan_new/zlib/1.2.11/_/_/build/bd500648356902bb5cb5a6961daa81fc25f3f22c/source_subfolder/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: E:/conan_new/zlib/1.2.11/_/_/build/bd500648356902bb5cb5a6961daa81fc25f3f22c/build_subfolder
zlib/1.2.11: Found Visual Studio 16
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.11
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

----Running------
> cmake --build E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c\build_subfolder -- -j4
-----------------
[16/16] Linking C static library lib\zlibstaticd.lib
zlib/1.2.11: Package 'bd500648356902bb5cb5a6961daa81fc25f3f22c' built
zlib/1.2.11: Build folder E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c
zlib/1.2.11: Generated conaninfo.txt
zlib/1.2.11: Generated conanbuildinfo.txt
zlib/1.2.11: Generating the package
zlib/1.2.11: Package folder E:\conan_new\zlib\1.2.11\_\_\package\bd500648356902bb5cb5a6961daa81fc25f3f22c
zlib/1.2.11: Calling package()
zlib/1.2.11: Found Visual Studio 16
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.11
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

----Running------
> cmake --build E:\conan_new\zlib\1.2.11\_\_\build\bd500648356902bb5cb5a6961daa81fc25f3f22c\build_subfolder --target install -- -j4
-----------------
[1/2] Install the project...
-- Install configuration: "Debug"
-- Installing: E:/conan_new/zlib/1.2.11/_/_/package/bd500648356902bb5cb5a6961daa81fc25f3f22c/lib/zlibstaticd.lib
-- Installing: E:/conan_new/zlib/1.2.11/_/_/package/bd500648356902bb5cb5a6961daa81fc25f3f22c/include/zconf.h
-- Installing: E:/conan_new/zlib/1.2.11/_/_/package/bd500648356902bb5cb5a6961daa81fc25f3f22c/include/zlib.h
ERROR: zlib/1.2.11: Error in package() method, line 127
        self._rename_libraries()
while calling '_rename_libraries', line 114
        tools.rename(current_lib, os.path.join(lib_path, "libzlib.a"))
        ConanException: rename E:\conan_new\zlib\1.2.11\_\_\package\bd500648356902bb5cb5a6961daa81fc25f3f22c\lib\libzlibstatic.a to E:\conan_new\zlib\1.2.11\_\_\package\bd500648356902bb5cb5a6961daa81fc25f3f22c\lib\libzlib.a failed: [WinError 2] The system cannot find the file specified: 'E:\\conan_new\\zlib\\1.2.11\\_\\_\\package\\bd500648356902bb5cb5a6961daa81fc25f3f22c\\lib\\libzlibstatic.a' -> 'E:\\conan_new\\zlib\\1.2.11\\_\\_\\package\\bd500648356902bb5cb5a6961daa81fc25f3f22c\\lib\\libzlib.a'
```

</details>

**Comment by ericLemanissier** [comment]
if you want to compile with clang-cl, then your profile must indicate it with
```
compiler=Visual Studio
compiler.toolset=ClangCL
```
on Windows, `compiler=clang` means you want to use regular clang

**Comment by ericLemanissier** [comment]
fixed by conan-io/conan-center-index#9807, this can be closed
